### PR TITLE
feat(slight-shim): mount the shim's resolv.conf to host's

### PR DESCRIPTION
**Pull request by Mossaka**
This PR mounts the shim's resolv.conf to host's, so that the shim can use host's nameserver to do dns resolution. A consequence of this PR is that we can use outbound-http in our shim, and thus each http handler can read/write to external services such as AWS DynamoDB.

### CODE REVIEW

**Comment by devigned** [comment]
@Mossaka I thought we hand DNS resolution working by chroot'ing to the rootfs dir. Was that not the case?

**Comment by cpuguy83** [comment]
We ran into an issue with that due to missing ca certs in the container rootfs.

**Comment by devigned** [comment]
> We ran into an issue with that due to missing ca certs in the container rootfs.

Would it be better to mount the CA certs or to include the CA certs in the image?

File: images/slight/src/main.rs [hunk 1]
```diff
@@ -34,19 +37,26 @@ fn handle_hello(req: Request) -> Result<Response, Error> {
...
-        body: request.body,
+        body: Some(value),
         status: 500,
     })
 }
 
 #[register_handler]
 fn handle_bar(request: Request) -> Result<Response, Error> {
     assert_eq!(request.method, Method::Put);
+    println!("request body: {:?}", request.body);
```
Comment (cpuguy83) [inline images/slight/src/main.rs:52]: Did you still want this here?

File: deployments/workloads/slight/workload.yaml [hunk 2]
```diff
@@ -15,8 +15,22 @@ spec:
       runtimeClassName: wasmtime-slight
       containers:
         - name: testwasm
-          image: docker.io/mossaka/slight-try:latest
+          image: docker.io/mossaka/slight-try3:awsdynamodb
```
Comment (devigned) [inline deployments/workloads/slight/workload.yaml:18]: We want to have an image reference similar to: https://github.com/deislabs/containerd-wasm-shims/blob/cb07e66d03625918cd37c22b93c6b760133f609e/deployments/workloads/spin/workload.yaml#L18.

File: images/slight/slightfile.toml [hunk 3]
```diff
@@ -1,4 +1,8 @@
 specversion = "0.1"
+secret_store = "configs.envvars"
 
 [[capability]]
 name = "http"
+
+[[capability]]
```
Comment (devigned) [inline images/slight/slightfile.toml:7]: For the simple case for an example workload, I think we would like to limit functionality to capabilities that do not need secrets.

**Review by devigned — CHANGES_REQUESTED** [review]

File: images/slight/slightfile.toml (hunk 3 above)
Comment (devigned) [reply images/slight/slightfile.toml:7]: Perhaps, use filesystem.

**Pull request merged**